datastore/noncemanager.go

```diff
@@ -87,7 +89,20 @@ func (db *DB) CreateDeviceNonce() (DeviceNonce, error) {
 	}
 
 	// Create the nonce in the database
-	_, err = db.Exec(createDeviceNonceSQL, nonce.Nonce, nonce.TimeStamp)
+	if Environ.Config.Driver == "sqlite3" {
+		// Need to generate our own ID
+		var nextID int
+		err = db.QueryRow(maxIDDeviceNonceSQLite).Scan(&nextID)
```

Comment:
not sure if this snippet taking the last inserted id is of help in this case:
```

// CreateQueueResource creates a new application in the datastore
func (d *DS) CreateQueueResource(res *QueueResource) (int, error) {
	result, err := d.Exec(insertQueueResourceSQL, res.Status, res.Location, res.Resources, res.Due.UTC())
	if err != nil {
		return -1, err
	}

	id, err := result.LastInsertId()
	return int(id), err
}
```

Reply:
This is a different use-case. With the serial-vault we have an auto-generated `id` in postgresql, but sqlite needs us to supply a value for the `id` field. What lastInsertId does is returns a hidden `rowid` field from the record... not quite the same thing